wrappers/golang/core/slice.go

```diff
@@ -54,6 +55,11 @@ func (d DeviceSlice) CheckDevice() {
 	}
 }
 
+func (d DeviceSlice) GetPointerSafe() unsafe.Pointer {
+	d.CheckDevice()
```

Comment:
I'm not sure this is needed here. Its possible to just use this function to pass the pointer around and not use it immediately.

Reply:
I think this should be removed.